src/librustc/mir/mod.rs

```diff
@@ -2096,6 +2096,15 @@ impl<'tcx> Place<'tcx> {
             Place::Base(base) => op(base, next.iter()),
         }
     }
+
+    /// Returns `true` if this is a promoted place.
+    pub fn is_promoted(&self) -> bool {
```

Comment:
I wouldn't add helpers for a temporary hack.

Reply:
Fixed.